tools/clang/lib/SPIRV/InitListHandler.cpp

```diff
@@ -0,0 +1,208 @@
+//===------- InitListHandler.cpp - Initializer List Handler -----*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//===----------------------------------------------------------------------===//
+//
+//  This file implements an initalizer list handler that takes in an
+//  InitListExpr and emits the corresponding SPIR-V instructions for it.
+//
+//===----------------------------------------------------------------------===//
+
+#include "InitListHandler.h"
+
+#include "llvm/ADT/SmallVector.h"
+
+namespace clang {
+namespace spirv {
+
+InitListHandler::InitListHandler(SPIRVEmitter &emitter)
+    : theEmitter(emitter), theBuilder(emitter.getModuleBuilder()),
+      typeTranslator(emitter.getTypeTranslator()),
+      diags(emitter.getDiagnosticsEngine()) {}
+
+uint32_t InitListHandler::process(const InitListExpr *expr) {
+  initializers.clear();
+  scalars.clear();
+
+  flatten(expr);
+
+  const uint32_t init = createInitForType(expr->getType());
+
+  /// We should have consumed all initializers and scalars extracted from them.
+  assert(initializers.empty());
+  assert(scalars.empty());
+
+  return init;
+}
+
+void InitListHandler::flatten(const InitListExpr *expr) {
+  const auto numInits = expr->getNumInits();
+
+  for (uint32_t i = 0; i < numInits; ++i) {
```

Comment:
https://clang.llvm.org/doxygen/classclang_1_1InitListExpr.html#a89b3c23b4c4d4acb99584139690f9962

Seems that our version of clang doesn't have it though. never mind then.